dttools/src/gpu_info.c

```diff
@@ -23,36 +23,20 @@ See the file COPYING for details.
 
 int gpu_info_get()
 {
-	int pipefd[2];
-	pipe(pipefd);
-
-	pid_t pid = fork();
-
-	if(pid<0) {
-		return 0;
-	} else if(pid==0) {
-		close(pipefd[0]);
-		dup2(pipefd[1], fileno(stdout));
-		char *args[] = {GPU_AUTODETECT, NULL};
-		if(!access(GPU_AUTODETECT, R_OK|X_OK)){
-			execv(GPU_AUTODETECT, args);
-		} else {
-			execvp(GPU_AUTODETECT, args);
-		}
-		_exit(0);
-	} else {
-		close(pipefd[1]);
-		int status = 0;
-		int gpu_count = 0;
-		char buffer[10]; /* Enough characters to hold a decimal representation of a 32 bit int. */
-		if(read(pipefd[0], buffer, 10)){
-			waitpid(pid, &status, 0);
-			gpu_count = atoi(buffer);
-		}
-
-		close(pipefd[0]);
-		return gpu_count;
+	char* nvidia_cmd= "/bin/nvidia-smi --query-gpu=count --format=csv,noheader";
+	if(access(nvidia_cmd, X_OK) != 0){	
+			return 0;
 	}
+	
+	FILE *pipe = popen(nvidia_cmd, "r");
+	if (!pipe)
+		return 0;
+	
+	int gpus;
+	fscanf(pipe, "%d", &gpus);	
```

Comment:
Make sure to check for the case that fscanf fails, so you can handle systems where nvidia-smi fails and the error message doesn't match "%d"